main/middleware/apisix_user_test.py

```diff
@@ -62,6 +62,65 @@ def test_get_request(mocker, mock_login):
     assert user.global_id == apisix_user_info["sub"]
 
 
+@pytest.mark.django_db(transaction=True)
+def test_get_request_existing_user_no_globalid(mocker, mock_login):
+    """Test that a valid request updates existing user with same email, no global_id."""
+    close_old_connections()
+    user = UserFactory.create(email=apisix_user_info["email"], global_id=None)
+    mock_request = mocker.Mock(
+        META={
+            "HTTP_X_USERINFO": b64encode(json.dumps(apisix_user_info).encode()),
+        },
+        user=AnonymousUser(),
+    )
+    apisix_middleware = ApisixUserMiddleware(mocker.Mock())
+    apisix_middleware.process_request(mock_request)
+    mock_login.assert_called_once()
+    user.refresh_from_db()
+    assert user.username == apisix_user_info["preferred_username"]
+    assert user.email == apisix_user_info["email"]
+    assert user.global_id == apisix_user_info["sub"]
+
+
+@pytest.mark.django_db(transaction=True)
+def test_get_request_existing_user_with_globalid(mocker, mock_login):
+    """Test that a valid request doesn't change global_id of existing user with same email"""
+    close_old_connections()
+    user = UserFactory.create(email=apisix_user_info["email"], global_id="abc123")
+    old_username = user.username
+    mock_request = mocker.Mock(
+        META={
+            "HTTP_X_USERINFO": b64encode(json.dumps(apisix_user_info).encode()),
+        },
+        user=AnonymousUser(),
+    )
+    apisix_middleware = ApisixUserMiddleware(mocker.Mock())
+    apisix_middleware.process_request(mock_request)
+    user.refresh_from_db()
+    assert user.username == old_username
+    assert user.global_id == "abc123"
+    assert user.email == apisix_user_info["email"]
+
+
+@pytest.mark.django_db(transaction=True)
+def test_get_request_existing_user_with_email(mocker, mock_login):
```

Comment:
I think this is the same as the first test. We do need a test though where the `global_id` matches and the email should be updated though.